spec/classes/minecraft_source_spec.rb

```diff
@@ -0,0 +1,45 @@
+require 'spec_helper'
+
+describe 'minecraft::source', :type => :class do
+  let :pre_condition do
+    'include ::minecraft'
+  end
+
+  let :facts do
+    {
+      :osfamily               => 'debian',
+      :operatingsystem        => 'ubuntu',
+      :operatingsystemrelease => '14.04',
+      :lsbdistcodename        => 'trusty',
+      :id                     => 'root',
+      :concat_basedir         => '/dne',
+      :path                   => '/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin',
+      :kernel                 => 'Linux',
+    }
+  end
+
+  it { is_expected.to compile }
+  it { is_expected.to contain_class('minecraft::source') }
+  it 'runs with defaults' do
+  end
+  context 'with defaults' do
```

Comment:
oh I see. Yeah, I'll do that.

-- Briggs

> On Jun 11, 2016, at 6:30 PM, Alex Fisher <notifications@github.com> wrote:
> 
> In spec/classes/minecraft_source_spec.rb:
> 
> > +      :osfamily               => 'debian',
> > +      :operatingsystem        => 'ubuntu',
> > +      :operatingsystemrelease => '14.04',
> > +      :lsbdistcodename        => 'trusty',
> > +      :id                     => 'root',
> > +      :concat_basedir         => '/dne',
> > +      :path                   => '/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin',
> > +      :kernel                 => 'Linux',
> > +    }
> > +  end
> > +
> > +  it { is_expected.to compile }
> > +  it { is_expected.to contain_class('minecraft::source') }
> > +  it 'runs with defaults' do
> > +  end
> > +  context 'with defaults' do
> I was thinking more about this being a 'private' class. Users aren't expected to declare it themselves and it depends on variables from init.pp. Since you can't actually test it in isolation, the tests you've added could probably be part of spec/minecraft_spec.rb?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
>